src/backend/sgx/attestation.rs

```diff
@@ -0,0 +1,372 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Credit to: https://github.com/fortanix/rust-sgx/tree/master/aesm-client
+// for examples of AESM Requests.
+
+use crate::protobuf::aesm_proto::{
+    Request, Request_GetQuoteExRequest, Request_GetSupportedAttKeyIDNumRequest,
+    Request_GetSupportedAttKeyIDsRequest, Request_InitQuoteExRequest, Response,
+    Response_GetQuoteExResponse, Response_InitQuoteExResponse,
+};
+
+use sallyport::syscall::{SGX_QUOTE_SIZE, SGX_TI_SIZE};
+
+use std::io::{Error, ErrorKind, Read, Write};
+use std::mem::size_of;
+use std::os::unix::net::UnixStream;
+use std::slice::{from_raw_parts, from_raw_parts_mut};
+
+use protobuf::Message;
+
+const AESM_SOCKET: &str = "/var/run/aesmd/aesm.socket";
+const TIMEOUT: u32 = 1_000_000;
+const SGX_KEY_ID_SIZE: u32 = 256;
+
+// Specifies the protobuf Request type to communicate with AESMD.
+#[derive(Debug)]
+enum ReqType {
+    TInfo,
+    KeySize,
+    Quote,
+}
+
+impl ReqType {
+    fn set_request(
+        &self,
+        report: Option<&[u8; 432]>,
+        akid: Option<Vec<u8>>,
+        size: Option<usize>,
+    ) -> Result<Request, Error> {
+        let mut req = Request::new();
+
+        match self {
+            ReqType::TInfo => {
+                let akid = match akid {
+                    Some(a) => a,
+                    None => {
+                        return Err(Error::new(
+                            ErrorKind::Other,
+                            "no attestation key ID provided for setting Init Quote Ex request",
+                        ));
+                    }
+                };
```

Comment:
How about something like this instead? https://doc.rust-lang.org/nightly/core/option/enum.Option.html#method.ok_or_else
```suggestion
                let akid = akid.ok_or_else(|| Error::new(
                               ErrorKind::Other,
                               "no attestation key ID provided for setting Init Quote Ex request",
                 ))?;
```